go/api/config.go

```diff
@@ -0,0 +1,257 @@
+	return config
+}
+
+// WithUseTLS configures the TLS settings for this configuration. Set to true if communication with
+// the cluster should use Transport Level Security. This setting should match the TLS configuration
+// of the server/cluster, otherwise the connection attempt will fail.
+func (config *RedisClientConfiguration) WithUseTLS(useTLS bool) *RedisClientConfiguration {
+	config.useTLS = useTLS
+	return config
+}
+
+// WithCredentials sets the credentials for the authentication process. If none are set, the client
+// will not authenticate itself with the server.
+func (config *RedisClientConfiguration) WithCredentials(credentials *RedisCredentials) *RedisClientConfiguration {
+	config.credentials = credentials
+	return config
+}
+
+// WithReadFrom sets the client's [ReadFrom] strategy. If not set, [PRIMARY] will be used.
+func (config *RedisClientConfiguration) WithReadFrom(readFrom ReadFrom) *RedisClientConfiguration {
+	config.readFrom = readFrom
+	return config
+}
+
+// WithRequestTimeout sets the duration in milliseconds that the client should wait for a request to
+// complete. This duration encompasses sending the request, awaiting for a response from the server,
+// and any required reconnections or retries. If the specified timeout is exceeded for a pending
+// request, it will result in a timeout error. If not set, a default value will be used.
+func (config *RedisClientConfiguration) WithRequestTimeout(requestTimeout uint32) *RedisClientConfiguration {
+	config.requestTimeout = int32(requestTimeout)
+	return config
+}
+
+// WithReconnectStrategy sets the [BackoffStrategy] used to determine how and when to reconnect, in
+// case of connection failures. If not set, a default backoff strategy will be used.
+func (config *RedisClientConfiguration) WithReconnectStrategy(strategy *BackoffStrategy) *RedisClientConfiguration {
+	config.reconnectStrategy = strategy
+	return config
+}
+
+// WithDatabaseId sets the index of the logical database to connect to.
+func (config *RedisClientConfiguration) WithDatabaseId(id uint32) *RedisClientConfiguration {
+	config.databaseId = int32(id)
+	return config
+}
+
+// RedisClusterClientConfiguration represents the configuration settings for a Cluster Redis client.
+// Notes: Currently, the reconnection strategy in cluster mode is not configurable, and exponential
+// backoff with fixed values is used.
+type RedisClusterClientConfiguration struct {
+	baseClientConfiguration
+}
+
+// NewRedisClusterClientConfiguration returns a [RedisClientConfiguration] with default
+// configuration settings. For further configuration, use the [RedisClientConfiguration] With*
+// methods.
+func NewRedisClusterClientConfiguration() *RedisClusterClientConfiguration {
+	return &RedisClusterClientConfiguration{
+		baseClientConfiguration: baseClientConfiguration{},
+	}
+}
+
+func (config *RedisClusterClientConfiguration) toProtobufConnRequest() *protobuf.ConnectionRequest {
```

Comment:
We shouldn't expose this function to the user

Reply:
yep, fields/methods that start with a lower-case character are not exported in Go, so users won't be able to access this